scripts/metavar.py

```diff
@@ -1204,6 +1205,47 @@ def add_to_list(self, vlist):
 
 ###############################################################################
 
+class VarUnitConv(VarAction):
+    """A class to perform unit conversions"""
+
+    def __init__(self, set_action, local_name, kind, **kwargs):
+        """Initialize unit conversion"""
+        self._local_name      = local_name
+        self._local_name_temp = local_name+'_local'
+        self._kind            = kind
+        for key, value in kwargs.items():
+            setattr(self, "_"+key, value)
+        self._set_action = set_action
+
+    def add_local(self, var):
+        """Add a local variable for unit conversion"""
+        tmp_var = var.clone(self._local_name_temp)
+        self.__group.manage_variable(tmp_var)
+
+    def write_action(self, vadict):
+        """Return unit conversion transformation"""
+        if self._set_action:
+            if self._key == 'from':
```

Comment:
While this works now, a problem with this design comes up with multiple variable actions necessary for one variable. For instance, flipping vertically and doing unit conversions, maybe adding a vloop around it. Trying to come up with a smart code generator that combines all this in one assigment is basically impossible, therefore either one after the other (which then requires other logic so that the rhs in the first assignment is `__local_name`, but all future rhs use `__local_name_temp`, and vice versa for the reverse process after the call to the subroutine), or generate nested function calls that recursively perform these actions. I think the former is easier to implement, but it probably won't allow you to substitute in the variable names in the var action expression until they get written out in lines 1218 / 1223 in scripts/suite_objects.py

Reply:
Excellent point. I didn't consider multiple actions on the same variable.
The way I do the write step is "by action", w/o consideration for other actions on that variable. But as you say, it should be "by variable", accounting for all actions on the variable.

Reply:
That's actually the major limitation in `ccpp_prebuild`, I worked around it by coding up all possible combinations (of which there are only four in prebuild, but in capgen there will be a lot more, therefore the need to do better).

Reply:
Gotcha. I'm starting to understand, albeit late to the party.